src/de/grobox/blitzmail/send/SendActivity.java

```diff
@@ -184,8 +186,18 @@ private void sendAttachment(@Nullable JSONObject jMail) {
 		}
 
 		try {
-			String num = String.valueOf(jMail.getJSONArray("attachments").length());
-			jMail.put("subject", num + " " + getString(R.string.files_shared) + " " + getString(R.string.app_name));
+			JSONArray files = jMail.getJSONArray(MAIL_ATTACHMENTS);
+			String filename = files.getJSONObject(0).getString("filename");
+			if (filename.length() > MAX_FILENAME_LENGTH) {
+				filename = filename.substring(0, MAX_FILENAME_LENGTH - 2) + "…";
+			}
+			if (files.length() == 1) {
+				jMail.put(MAIL_SUBJECT, getString(R.string.subject_single_file, filename));
+			} else {
```

Comment:
Something I notice just now. Can it happen that `files.length() == 0` for some reason?

Reply:
I couldn't find an easy way to get 0 because even if an attachment fails it would still be put in attachments.
However I have made changes to hopefully cover any possibility of attachments being empty or null or missing from the jMail object.

(I've had some commit issues hopefully my changes went through.)